no-ag.js

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env node
+
+import { glob, readFile, stat } from 'node:fs/promises';
+
+const phrases = [
+  "require('punycode')",
+  'require("punycode")',
+  "import('punycode')",
+  'import("punycode")',
+  "from 'punycode'",
+  'from "punycode"',
+  // side-effects -- hopefully these are very rare
+  "import 'punycode'",
+  'import "punycode"',
+];
+
+async function punycodeUsages() {
+  console.info('Scanning for punycode usages, this may take a moment.');
+
+  const allJsFilesIterator = await glob(['**/*.js', '**/*.cjs', '**/*.mjs'], {
+    exclude: () => false
+  });
+
+  const filesWithPunycode = new Set();
+
+  for await (const file of allJsFilesIterator) {
+    if (file.includes('punycode-detector')) {
+      continue;
+    }
+
+    const fileStats = await stat(file);
+
+    if (fileStats.isDirectory()) {
+      continue;
+    }
+
+    console.clear();
+
+    console.info('Scanning for punycode usages, this may take a moment.');
+
+    console.info(`Scanning file: ${file}`);
+
+    const content = await readFile(file, 'utf-8');
+    const doesContentContainPunycode = phrases.find((phrase) => content.includes(phrase));
+
+    if (doesContentContainPunycode) {
+      filesWithPunycode.add(file);
+    }
+  }
+
+  return filesWithPunycode;
+}
+
+const usages = await punycodeUsages();
+
+console.clear();
```

Comment:
Note: I added a bunch of these folks in my version, you might want to have it streamlined across versions :)